pkg/export/export.go

```diff
@@ -416,63 +442,103 @@ const (
 )
 
 // ApplyConfig updates the exporter state to the given configuration.
-// Must be called at least once before Export() can be used.
-func (e *Exporter) ApplyConfig(cfg *config.Config) (err error) {
-	// If project_id, location, or cluster were set through the external_labels in the config file,
-	// these values take precedence. If they are unset, the flag value, which defaults to an
-	// environment-specific value on GCE/GKE, is used.
-	builder := labels.NewBuilder(cfg.GlobalConfig.ExternalLabels)
-
-	if !cfg.GlobalConfig.ExternalLabels.Has(KeyProjectID) {
-		builder.Set(KeyProjectID, e.opts.ProjectID)
-	}
-	if !cfg.GlobalConfig.ExternalLabels.Has(KeyLocation) {
-		builder.Set(KeyLocation, e.opts.Location)
+func (e *Exporter) ApplyConfig(cfg *config.Config, opts *ExporterOpts) (err error) {
+	if e.isNopExporter() {
```

Comment:
Do we really need testing code be in prod code though? It's just assertion for test generally?

Reply:
Thanks, I got rid of this and added a comment instead.